File: javascript/leetcode/172. Factorial Trailing Zeroes.js
/**
 * @param {number} n
 * @return {number}
 */
var trailingZeroes = function (n) {
  let fives = 0;

  while (n >= 5) {
    n = parseInt(n / 5);
    fives += n;
  }

  return fives;
};

File: c++/leetcode/172. Factorial Trailing Zeroes.cpp
class Solution
{
public:
  int trailingZeroes(int n)
  {
    int number_of_fives = 0;

    while (n >= 5)
    {
      n /= 5;
      number_of_fives += n;
    }

    return number_of_fives;
  }
};